File: Leetcode 3101-3200/3120/3120.java
// Leetcode 3120: Count the Number of Special Characters I
// https://leetcode.com/problems/count-the-number-of-special-characters-i/
// Solved on 26th of May, 2026
class Solution {
    /**
     * Counts the number of special characters in a string.
     * A character is special if both its lowercase and uppercase forms appear in the string.
     * @param word The input string consisting of English letters.
     * @return The number of unique characters that appear in both cases.
     */
    public int numberOfSpecialChars(String word) {
        int lowerMask = 0;
        int upperMask = 0;
        int length = word.length();
        for (int i = 0; i < length; i++) {
            char ch = word.charAt(i);
            if (ch >= 'a' && ch <= 'z') {
                lowerMask |= (1 << (ch - 'a'));
            } else {
                upperMask |= (1 << (ch - 'A'));
            }
        }
        return Integer.bitCount(lowerMask & upperMask);
    }
}

File: Leetcode 3101-3200/3120/3120.py
# Leetcode 3120: Count the Number of Special Characters I
# https://leetcode.com/problems/count-the-number-of-special-characters-i/
# Solved on 26th of May, 2026
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        """
        Counts the number of special characters in a string.
        A character is special if both its lowercase and uppercase forms appear in the string.

        :param word: The input string to check for special characters.
        :return: The total count of unique special characters.
        """
        lowerMask = 0
        upperMask = 0

        for currentLetter in word:
            if currentLetter.islower():
                lowerMask |= 1 << (ord(currentLetter) - 97)
            else:
                upperMask |= 1 << (ord(currentLetter) - 65)

        return bin(lowerMask & upperMask).count('1')